### PR TITLE
Refactor `tx_pool` state computation

### PR DESCRIPTION
**Pull request by lucafavatella**
Please refer to commit messages for details.

Refactoring performed while working on #255

### CODE REVIEW

File: apps/ae_core/src/consensus/tx_pool.erl [hunk 1]
```diff
@@ -91,17 +91,12 @@ code_change(_OldVsn, State, _Extra) ->
...
-	    state2(PrevHeader);
-	_ ->
-	    #f{trees = block:trees(Block),
-	       height = block:height(Block)}
-    end.
+    B = block:top(),
+    state_from_block(B).
+
+state_from_block(B) ->
+    #f{trees = block:trees(B),
```
Comment (sennui) [inline apps/ae_core/src/consensus/tx_pool.erl]: record f ?

File: apps/ae_core/src/consensus/chain/block.erl [hunk 2]
```diff
@@ -123,15 +123,16 @@ get_by_hash(H) ->
 
 -spec top() -> block().
 top() ->
-    TH = headers:top(),
-    top(TH).
+    most_recent_observed_block_in_chain(headers:top()).
 
-top(Header) ->
+-spec most_recent_observed_block_in_chain(Chain::headers:header()) -> block().
+most_recent_observed_block_in_chain(Header) ->
```
Comment (sennui) [inline apps/ae_core/src/consensus/chain/block.erl]: i do like explicit names, but top seems to be standard for blockchain... so the long name paradoxally may be more confusing

File: apps/ae_core/src/consensus/tx_pool.erl (hunk 1 above)
Comment (lucafavatella) [inline apps/ae_core/src/consensus/tx_pool.erl]: Ok - will rename to state.

File: apps/ae_core/src/consensus/chain/block.erl (hunk 2 above)
Comment (lucafavatella) [inline apps/ae_core/src/consensus/chain/block.erl]: What I was finding confusing was that it is not the block corresponding to the top header, but the most recent **observed** block starting from the top header. Will rename back to top - adding comment.

**Pull request merged**